index.js

```diff
@@ -593,69 +387,57 @@ class XiaomiRoborockVacuum {
-        this.services.info.setCharacteristic(
-          Characteristic.FirmwareRevision,
-          `${firmware}`
-        );
+        this.services.info.setCharacteristic(Characteristic.FirmwareRevision, `${firmware}`);
         this.log.info(`STA getDevice | Firmwareversion: ${firmware}`);
       } catch (err) {
         this.log.error(`ERR getDevice | miIO.info | ${err}`);
       }
-
-      this.device.on("errorChanged", (error) => this.changedError(error));
-      this.device.on("stateChanged", (state) => {
-        if (state.key === "cleaning") {
+      //Changed by Mai
+      var minStep = this.findSpeedModes().speed.length == 6 ? 20 : 25;
+      this.log.info(`INF initializeDevice | ${this.model} ${this.firmware} | Setting the minimal step for fan service to ${minStep}.`);
+      this.services.fan
+        .getCharacteristic(Characteristic.RotationSpeed).setProps({
+          minStep: minStep
+        }).on('get', (cb) => callbackify(() => this.getSpeed(), cb))
+        .on('set', (newState, cb) => callbackify(() => this.setSpeed(newState), cb));
+
+      this.device.on('errorChanged', (error) => this.changedError(error));
+      this.device.on('stateChanged', (state) => {
+        this.log.info(`INF stateChanged | ${this.model} | New state: ${state.key}=${state.value}`);
+        if (state.key === 'cleaning') {
           this.changedCleaning(state.value);
           this.changedPause(state.value);
-        } else if (state.key === "charging") {
+        } else if (state.key === 'charging') {
           this.changedCharging(state.value);
-        } else if (state.key === "fanSpeed") {
+        } else if (state.key === 'fanSpeed') {
           this.changedSpeed(state.value);
-        } else if (state.key === "batteryLevel") {
+        } else if (state.key === 'batteryLevel') {
           this.changedBattery(state.value);
         } else {
-          this.log.debug(
-            `DEB stateChanged | ${this.model} | Not supported stateChanged event: ${state.key}:${state.value}`
-          );
+          this.log.WARN(`WRN stateChanged | ${this.model} | Not supported stateChanged event: ${state.key}=${state.value}`);
         }
       });
 
-      // Now that we know the model, amend the steps in the Rotation speed (for better usability)
-      const minStep = 100 / (this.findSpeedModes().speed.length - 1);
-      this.services.fan
-        .getCharacteristic(Characteristic.RotationSpeed)
-        .setProps({ "minStep": minStep });
-
       await this.getState();
       // Refresh the state every 30s so miio maintains a fresh connection (or recovers connection if lost until we fix https://github.com/nicoh88/homebridge-xiaomi-roborock-vacuum/issues/81)
       clearInterval(this.getStateInterval);
-      this.getStateInterval = setInterval(
-        () => this.getState(),
-        GET_STATE_INTERVAL_MS
-      );
+      //Changed by Mai
+      this.getStateInterval = setInterval(() => this.device.poll(false), GET_STATE_INTERVAL_MS);
```

Comment:
Actual change to include in a PR (although already changed in #188).

Reply:
This is a tricky one, the poll method gets the status of the robot and cache values in properties, it is a better way to get period state update. It's more responsive than call getState.